### PR TITLE
Add validation pipeline for mooncake

### PR DESCRIPTION
**Pull request by jfaurskov**
# Overview/Summary

Adds validate pipeline for mooncake

## This PR fixes/adds/changes/removes

1. Add tests/pipelines/mc-base-unit-validate.yml

### Breaking Changes

None

## Testing Evidence

To be updated

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [x] *(ALZ Bicep Core Team Only)* Associated it with relevant [116484](https://dev.azure.com/unifiedactiontracker/Solution%20Engineering/_workitems/edit/116484)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [ ] Performed testing and provided evidence.
- [x] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [ ] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

File: tests/pipelines/mc-base-unit-validate.yml [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+variables:
+  - group: csu-bicep-environment-mc
+    #Add subscriptionName and subscriptionId to keyvault as subscriptionName and subscriptionId and include in variable group
+    #verify mctenant, mcLocation in key vault and variable group
+  - name: ResourceGroupName
+    #value: "rsg-github-pr-$(System.PullRequest.PullRequestNumber)"
+    value: "rsg-github-pr-233"
+  - name: ManagementGroupPrefix
+    #value: "PR-$(System.PullRequest.PullRequestNumber)"
+    value: "PR-233"
```
Comment (jtracey93) [inline tests/pipelines/mc-base-unit-validate.yml:14]: think you need to flip out the commented-out values from your testing 😁

File: tests/pipelines/mc-base-unit-validate.yml [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+trigger: none
+
+variables:
+  - group: csu-bicep-environment-mc
+    #Add subscriptionName and subscriptionId to keyvault as subscriptionName and subscriptionId and include in variable group
+    #verify mctenant, mcLocation in key vault and variable group
+  - name: ResourceGroupName
+    #value: "rsg-github-pr-$(System.PullRequest.PullRequestNumber)"
+    value: "rsg-github-pr-233"
```
Comment (jtracey93) [inline tests/pipelines/mc-base-unit-validate.yml:11]: think you need to flip out the commented-out values from your testing 😁

File: tests/pipelines/mc-base-unit-validate.yml [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  - name: ManagementGroupPrefix
+    #value: "PR-$(System.PullRequest.PullRequestNumber)"
+    value: "PR-233"
+  - name: TopLevelManagementGroupDisplayName
+    #value: "PR $(System.PullRequest.PullRequestNumber) Azure Landing Zones"
+    value: "PR 233 Azure Landing Zones"
```
Comment (jtracey93) [inline tests/pipelines/mc-base-unit-validate.yml:17]: think you need to flip out the commented-out values from your testing 😁

File: .github/scripts/Wipe-AlzTenant.ps1 [hunk 4]
```diff
@@ -22,44 +22,44 @@ $StopWatch.Start()
...
 $subDeployments | ForEach-Object -Parallel {
-    Write-Information "Removing $($_.DeploymentName) ..."
+    Write-Output "Removing $($_.DeploymentName) ..."
     Remove-AzSubscriptionDeployment -Id $_.Id
 }
 
 
 # Get all AAD Tenant level deployments
-$tenantDeployments = Get-AzTenantDeployment
+$tenantDeployments = Get-AzTenantDeployment | Where-Object {$PSItem.DeploymentName -like "$intermediateRootGroupID*"}
```
Comment (jtracey93) [inline .github/scripts/Wipe-AlzTenant.ps1:56]: is this going to break the clean-ups in the public cloud as they dont have these deployment names yet?

File: .github/scripts/Wipe-AlzTenant.ps1 [hunk 5]
```diff
@@ -22,44 +22,44 @@ $StopWatch.Start()
...
 
-Write-Information "Removing all Tenant level deployments"
+Write-Output "Removing all Tenant level deployments"
```
Comment (jtracey93) [inline .github/scripts/Wipe-AlzTenant.ps1:58]: makes this a bit more accurate around the deployment names?

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Looking good, just a few bits @jfaurskov

File: .github/scripts/Wipe-AlzTenant.ps1 (hunk 4 above)
Comment (jfaurskov) [reply .github/scripts/Wipe-AlzTenant.ps1:56]: It is actually, good catch. I'll rename to mc-Wipe-Alz-Tenant.ps1 and we can merge once we get the naming sorted.

**Review by jtracey93 — APPROVED** [review]
LGTM. @jfaurskov don't forget to do the ADO piece now the pipeline will be in the main repo 😁

**Pull request merged**